### PR TITLE
tsp, add test for cadl-ranch Type/Array

### PR DESCRIPTION
**Pull request by v-hongli1**
relate to [issue#2829](https://github.com/Azure/autorest.java/issues/2829)

Add test case for Type/Array/NullableBooleanValue, Type/Array/NullableInt32Value, Type/Array/NullableModelValue, Type/Array/NullableStringValue

### CODE REVIEW

File: typespec-tests/src/test/java/com/type/array/NullableBooleanValueClientTest.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+package com.type.array;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class NullableBooleanValueClientTest {
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/test/java/com/type/array/NullableBooleanValueClientTest.java:13]: public

File: typespec-tests/src/test/java/com/type/array/NullableBooleanValueClientTest.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    NullableBooleanValueClient client = new ArrayClientBuilder().buildNullableBooleanValueClient();
+
+    @Test
+    void get() {
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/test/java/com/type/array/NullableBooleanValueClientTest.java:18]: public

File: typespec-tests/src/test/java/com/type/array/NullableBooleanValueClientTest.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class NullableBooleanValueClientTest {
+
+    NullableBooleanValueClient client = new ArrayClientBuilder().buildNullableBooleanValueClient();
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/test/java/com/type/array/NullableBooleanValueClientTest.java:15]: private final

Comment (v-hongli1) [reply typespec-tests/src/test/java/com/type/array/NullableBooleanValueClientTest.java]: Fixed in the new version.

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by haolingdong-msft — APPROVED** [review]

**Pull request merged**